test/functional/tests/stress/test_stress_io_small_device.py

```diff
@@ -0,0 +1,107 @@
+
+    with TestRun.step(f"Start cache with {core_number} cores."):
+        cache = casadm.start_cache(cache_dev, cache_mode, cache_line_size, force=True)
+        cores = []
+        for i in range(core_number):
+            cores.append(cache.add_core(core_dev.partitions[i]))
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 1:
+            TestRun.fail(f"Expected caches count: 1; Actual caches count: {caches_count}.")
+        cores_count = len(casadm_parser.get_cores(cache.cache_id))
+        if cores_count != core_number:
+            TestRun.fail(f"Expected cores count: {core_number}; Actual cores count: {cores_count}.")
+
+    if cleaning_policy is not None:
+        with TestRun.step("Set cleaning policy."):
+            cache.set_cleaning_policy(cleaning_policy)
+
+    with TestRun.step(f"Stress cache for {int(stress_time.total_seconds() / 60)} minutes."):
+        for core in cores:
+            Fio().create_command() \
+                .io_engine(IoEngine.libaio) \
+                .io_depth(128) \
+                .direct() \
+                .time_based() \
+                .run_time(stress_time) \
+                .read_write(ReadWrite.randrw) \
+                .block_size(cache_line_size) \
+                .target(f"{core.system_path}") \
+                .run()
+
+    with TestRun.step("Count md5 sum for cache's exported object"):
+        md5sum_core_dev = []
+        for core in cores:
+            md5sum_core_dev.append(TestRun.executor.run(
+                f"md5sum -b {core.system_path}").stdout.split(" ")[0])
+
+    with TestRun.step("Stop cache."):
+        cache.stop()
+
+    with TestRun.step("Count md5 sum for core device"):
+        md5sum_cache_dev = []
+        for core_dev in core_dev.partitions:
+            md5sum_cache_dev.append(TestRun.executor.run(
+                f"md5sum -b {core_dev.system_path}").stdout.split(" ")[0])
+
+    with TestRun.step("Compare md5 sum of every cache's exported object and core."):
+        for i in range(core_number):
+            if md5sum_core_dev[i] != md5sum_cache_dev[i]:
+                TestRun.fail(f"Md5 sums {md5sum_core_dev[i]} of core device "
+                             f"and {md5sum_cache_dev[i]} of cache device are different.")
+
+    with TestRun.step("Stop all caches"):
+        casadm.stop_all_caches()
+
+
+@pytest.mark.parametrize("cleaning_policy", CleaningPolicy)
+@pytest.mark.parametrize("cache_mode", [CacheMode.WB, CacheMode.WO])
+@pytest.mark.parametrize("core_number", [1, 4])
+@pytest.mark.parametrize("cache_line_size", CacheLineSize)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_stress_small_cas_device_wb_wo_mode(cache_mode, cache_line_size,
+                                            core_number, cleaning_policy):
+    test_stress_small_cas_device(cache_mode, cache_line_size, core_number, cleaning_policy)
```

Comment:
Calling one test case from another looks weird. How about making it separate function and calling it from both test cases with appropriate params?

Reply:
putting steps inside a method will affect the ability to parse test steps properly (and we aim for that)
I wonder if we can make single test and parametrize like ("cache_mode_cleaning_policy", [(WB, [CleaningPolicy]), (WT, None), (WO, [CleaningPolicy]), ...]

Reply:
As long as all the test logic is contained within steps there is no problem with parsing, as we can prepare special test execution mode that skips actual code block within test step. That would produce simple log containing test step titles as a result of simplified test execution, and so having steps in separate function is not a problem (as long as it's not called from test step, but that would be equivalent of having nested test steps which is illegal anyway).

That kind of composite parametrization you proposed is technically doable, but if one day we decide to make some kind of requirement <-> test configuration mapping we're gonna have hard time with that.